File: 37-audit-logs/python-sdk/audit.py
import os, requests
API = "https://cost-katana-backend.store/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def get_audit_logs():
    res = requests.get(f"{API}/audit-logs",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"📋 Found {len(res.json()['data'])} audit logs")
    return res.json()["data"]

def search_logs(action, from_date=None):
    res = requests.post(f"{API}/audit-logs/search",
        json={"action": action, "from": from_date},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"🔍 Search results: {len(res.json()['data'])}")
    return res.json()["data"]

File: 36-audit-logs/npm-package/audit.ts
import axios from 'axios';
const API = 'https://api.costkatana.com/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function getAuditLogs() {
  const res = await axios.get(`${API}/audit-logs`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('📋 Found', res.data.data.length, 'audit logs');
  return res.data.data;
}

export async function searchLogs(action: string, from?: string) {
  const res = await axios.post(`${API}/audit-logs/search`, {action, from},
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('🔍 Search results:', res.data.data.length);
  return res.data.data;
}
